artemis/modules/nuclei.py

```diff
@@ -457,15 +481,21 @@ def run_multiple(self, tasks: List[Task]) -> None:
         self.log.info(f"running {len(templates)} templates and {len(self._workflows)} workflow on {len(tasks)} hosts.")
 
         targets: List[str] = []
-        for task in tasks:
-            targets.append(get_target_url(task))
-
         scan_groups = group_targets_by_missing_tech(targets, self.log)
         found_targets_after_grouping = []
         for scan_group in scan_groups.values():
             found_targets_after_grouping.extend(scan_group)
         assert set(found_targets_after_grouping) == set(targets)
 
+        for template_name, template_data in DAST_SCANNING.items():
+            DAST_SCANNING[template_name]["targets"] = list()
```

Comment:
I don't think modifying a global variable is a good idea, what about keeping DAST_SCANNING constant and collecting the target in a different one?